film_loader.py

```diff
@@ -0,0 +1,123 @@
+import sqlite3
+import json
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk
+
+
+def extract():
+    """
+    extract data from sql-db
+    :return:
+    """
+    connection = sqlite3.connect("db.sqlite")
+    cursor = connection.cursor()
+
+    # РќР°РІРµСЂРЅСЏРєР° СЌС‚Рѕ РїРёР»РёС‚СЃСЏ РІ РѕРґРёРЅ sql - Р·Р°РїСЂРѕСЃ, РЅРѕ РјРЅРµ РєР°Рє-С‚Рѕ Р»РµРЅРёРІРѕ)
+
+    # РџРѕР»СѓС‡Р°РµРј РІСЃРµ РїРѕР»СЏ РґР»СЏ РёРЅРґРµРєСЃР°, РєСЂРѕРјРµ СЃРїРёСЃРєР° Р°РєС‚РµСЂРѕРІ Рё СЃС†РµРЅР°СЂРёСЃС‚РѕРІ, РґР»СЏ РЅРёС… С‚РѕР»СЊРєРѕ id
+    cursor.execute("""
+        select id, imdb_rating, genre, title, plot, director,
+        -- comma-separated actor_id's
+        (
+            select GROUP_CONCAT(actor_id) from
+            (
+                select actor_id
+                from movie_actors
+                where movie_id = movies.id
+            )
+        ),
+
+        max(writer, writers)
+        from movies
+    """)
+
+    raw_data = cursor.fetchall()
+
+    # cursor.execute('pragma table_info(movies)')
+    # pprint(cursor.fetchall())
+
+    # РќСѓР¶РЅС‹ РґР»СЏ СЃРѕРѕС‚РІРµС‚СЃРІРёСЏ РёРґРµРЅС‚РёС„РёРєР°С‚РѕСЂР° Рё С‡РµР»РѕРІРµРєРѕС‡РёС‚Р°РµРјРѕРіРѕ РЅР°Р·РІР°РЅРёСЏ
+    actors = {row[0]: row[1] for row in cursor.execute('select * from actors where name != "N/A"')}
+    writers = {row[0]: row[1] for row in cursor.execute('select * from writers where name != "N/A"')}
+
+    return actors, writers, raw_data
+
+
+def transform(__actors, __writers, __raw_data):
+    """
+
+    :param __actors:
+    :param __writers:
+    :param __raw_data:
+    :return:
+    """
+    documents_list = []
+    for movie_info in __raw_data:
+        # Р Р°Р·С‹РјРµРЅРѕРІР°РЅРёРµ СЃРїРёСЃРєР°
+        movie_id, imdb_rating, genre, title, description, director, raw_actors, raw_writers = movie_info
+
+        if raw_writers[0] == '[':
+            parsed = json.loads(raw_writers)
+            new_writers = ','.join([writer_row['id'] for writer_row in parsed])
+        else:
+            new_writers = raw_writers
+
+        writers_list = [(writer_id, __writers.get(writer_id)) for writer_id in new_writers.split(',')]
```

Comment:
Так же посмотрел бы на NamedTuple. Он дает тебе возможность оперировать не индексами, а читаемыми атрибутами, даже при работе с tuple. И типизация улучшится.
```suggestion
        class WriterInfo(NamedTuple): ...
        writers_list = [WriterInfo(writer_id, __writers.get(writer_id)) for writer_id in new_writers.split(',')]
```